rust/main/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -677,14 +679,21 @@ async fn confirm_lander_task(
             .map(|(op, result)| async {
                 let message_id = op.id();
                 match result {
-                    Ok(Some(payload_id)) => Some((op, entrypoint.payload_status(payload_id).await)),
-                    Ok(None) | Err(_) => {
+                    Ok(Some(ids)) if !ids.is_empty() => {
+                        let op_futures = ids
+                            .into_iter()
+                            .map(|id| entrypoint.payload_status(id))
+                            .collect::<Vec<_>>();
+                        let op_results = join_all(op_futures).await;
```

Comment:
noting that this join_all will block synchronously on rocksdb under the hood, for each payload - but nothing we can improve about this